Fishing Plugin: Duplicate fishing icons fix

nrodriguesdev opened the pull request:
This bug fix addresses #1392.

**Problem:** The fishing plugin doesn't keep track of the fishing spots near the player. As a result, it tries to draw multiple icons in the same spot whenever fishing spots are stacked. 

**Fix:** I implemented 2 HashMaps to keep track of core information regarding each fishing spot. The logical height is no longer used to determine the height of text/icons. Instead, a fixed value (derived from the logical height of a fishing spot) is used to avoid odd logical heights caused by stacked spots.

![image](https://user-images.githubusercontent.com/25905465/52918820-9a4c0780-32c9-11e9-8b48-eea36c3cbc3c.png)

**Concerns**: I'm not confident that using two HashMaps is the right approach. Any feedback on this will be appreciated.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java:
```diff
@@ -57,6 +61,9 @@
 	private final Client client;
 	private final ItemManager itemManager;
 
+	private static HashMap<WorldPoint, HashMap<FishingSpot, NPC>> SPOTS_LIST = new HashMap<>();
```
Just make new type and use 1 map

nrodriguesdev replied:
I've gone ahead and created a new inner class to take care of that. Each instance of the class FishingSpotInfo will store 3 ArrayLists, containing a list of FishingSpots, NPCs, and counts. The getters were implemented using indexes to maintain the same lookup speed as before.

I've tested these new changes and they're functioning as expected.

Digbigpig commented on runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java:
```diff
@@ -123,37 +158,139 @@ public Dimension render(Graphics2D graphics)
+	}
 
-			if (config.showSpotNames())
+	private class FishingSpotInfo
```
Move this to it's own file

nrodriguesdev replied:
I can certainly do that. In regards to design, I assume you recommend moving the class to its own file because it contains more than basic getters/setters?

Digbigpig replied:
If you can, get on the RuneLite discord and talk with the guys in the #development chat channel. It sounds like they would like different/ smaller solution to this, and I wouldn't want you to waste your time.

As far as the separate file, I think it is just convention with this project to keep everything separate. The purpose for that might be to enforce modularity and make things obvious for other developers, but don't quote me on it.

nrodriguesdev commented:
Thanks for the feedback everyone. I've followed the advice of others on Discord. 

I've removed the new class and replaced it with a Multiset. This has reduced the number of lines of code and made it easier to read. 

I added this nasty bit to convert the Multiset(NPC) to Multiset(FishingSpot)
```
			Multiset<FishingSpot> spotList = HashMultiset.create();
			for (NPC npc : npcArray)
			{
				FishingSpot spot = FishingSpot.getSPOTS().get(npc.getId());
				spotList.add(spot);
			}
```
If there's a better way to do this, then please let me know. Thanks for checking this PR out and having the patience to deal with me lol.

xDemoN commented:
@Adam- you superseded this with 9fc04bf
